### DISCUSSION TITLE
Introduce a stepper

### DISCUSSION THREAD

**Discussion by Noah-Haf** [heroui-inc/heroui#1006] [Feature requests]
![image](https://user-images.githubusercontent.com/61884850/212197885-c5e92c37-c85e-4a29-8778-34b7dbc5a0b4.png)

Something like this, but in the style of this library.

**Comment by NickNaskida** [comment]
Yes, please add something like this. See the example below.

examples: https://ant.design/components/steps

**Comment by itsEricWu** [comment]
+1

**Comment by fvcoder** [comment]
+1

**Comment by andrewnguyen-wsu** [comment]
+1

**Comment by DaywisonSilva** [comment]
+1

**Comment by danbcd** [comment]
+1

**Comment by J4v4Scr1pt** [comment]
Fyi, there is a stepper planned for the Pro: https://feedback.nextui.pro/p/stepper

**Comment by eglove** [comment]
Well +1 to the Pro then. If this is doable with current components to the extent something like Ant does, I'd love to see it Pro.

If it makes more sense as a standalone component. Standalone component please. :)

**Comment by rayaanr** [comment]
I found that the stepper is available in the Pro version. In my opinion stepper is a basic component and most of like to have it in the free component library @jrgarciadev . I hope you'll consider about this.

**Comment by ngocnguyenthework** [comment]
+1

**Comment by ghost** [comment]
+1

**Comment by henriqueNectaco** [comment]
+1

**Reply by eglove** [reply]
Steppers are in pro now.

**Comment by DamiAdesola** [comment]
+1

**Comment by ProgrammerCrypto** [comment]
+1

**Comment by createdbymahmood** [comment]
+1

**Comment by Ejdyz** [comment]
+1

**Comment by Pszz** [comment]
+1

I think PRO should be a more refined and customized component, 
and the basic components of the UI library should be more complete.

**Comment by umar-mq** [comment]
+1

**Comment by bricevrs** [comment]
+1

**Comment by ghost** [comment]
+1

**Comment by J4v4Scr1pt** [comment]
Fyi:
There is a [free version](https://www.heroui.pro/components/application/steppers#component-simple-stepper) of the stepper in Pro.

**Comment by webmaasteriv** [comment]
+1

**Comment by numanaral** [comment]
Hey all, need to build one for a side project, here is an early version that can be of use for starters!

Comes with a hook and overridable options, and see the `Stepper.tsx` for the example usage!

<img width="712" height="733" alt="stepper" src="https://github.com/user-attachments/assets/fd4b23ac-9c6d-48bc-8dfa-3de3736fff79" />

https://gist.github.com/numanaral/e773fedb413515224c68b581e78b6e1d
